### PR TITLE
Introduce api to fetch investigations from server

### PR DESCRIPTION
**Pull request by rugpanov**

### CODE REVIEW

File: teamcity-rest-client-api/src/main/kotlin/org/jetbrains/teamcity/rest/api.kt [hunk 1]
```diff
@@ -164,6 +165,11 @@ interface BuildLocator {
     fun withAnyStatus(): BuildLocator
 }
 
+interface InvestigationLocator {
+    fun limitResults(count: Int): InvestigationLocator
+    fun all(): Sequence<Investigation>
```
[the remark this answers is not in the document]
Comment (rugpanov) [reply teamcity-rest-client-api/src/main/kotlin/org/jetbrains/teamcity/rest/api.kt]: Yes, for my reasons I would like to filter investigations for different projects, target types and resolve methods.

File: teamcity-rest-client-api/src/main/kotlin/org/jetbrains/teamcity/rest/api.kt [hunk 2]
```diff
@@ -450,6 +460,15 @@ interface Build {
     val finishDate: Date?
 }
 
+interface Investigation {
+    val id: InvestigationId
+    val state: InvestigationState
+    val assigneeUsername: String
+    val reporterUsername: String?
+    val comment: String
+    val removeMethod: InvestigationRemoveMethod
```
[the remark this answers is not in the document]
Comment (rugpanov) [reply teamcity-rest-client-api/src/main/kotlin/org/jetbrains/teamcity/rest/api.kt]: I tried to find proper word for the action. ResolveMethod is better?

[the remark this answers is not in the document]
Comment (chashnikov) [reply teamcity-rest-client-api/src/main/kotlin/org/jetbrains/teamcity/rest/api.kt]: Yes, `resolveMethod` looks better.

**Comment by chashnikov** [comment]
Merged, thank you for the contribution!

**Pull request merged**